lib/galaxy/managers/workflows.py

```diff
@@ -1337,6 +1336,8 @@ def _workflow_to_dict_export(self, trans, stored=None, workflow=None, internal=F
                 "label": step.label or None,
                 "annotation": annotation_str,
             }
+            if step.type == "tool":
+                step_dict["tool_id"] = step.tool_id
```

Comment:
This looks safer:
```suggestion
                step_dict["tool_id"] = content_id
```

Reply:
I disagree, use of `content_id ` is what got us here. That's explicit and and exactly the same as the first branch of the content_id property.

Reply:
We'd rewrite this for pydantic anyway and use the orm_mode, so that's also closer to what we'd do there.

Reply:
What I mean is that `ToolModule.get_content_id()` [returns](https://github.com/galaxyproject/galaxy/blob/9efb749830b2331b16cb30d88e4ff08dded49e4e/lib/galaxy/workflow/modules.py#L1573) `self.tool.id if self.tool else self.tool_id` which is not always the same as `step.tool_id` .

Reply:
That's not correct for exports.